src/main/kotlin/org/rust/lang/core/resolve/RustResolveEngine.kt

```diff
@@ -150,6 +155,19 @@ object RustResolveEngine {
         return resolveIn(ResolveScopeVisitor(ref, visited), enumerateScopesFor(ref))
     }
 
+    private fun resolveImportSpec(ref: RustImportSpec, visited: MutableSet<RustUseItem>): ResolveResult {
+        val basePath = ref.parentOfType<RustUseItem>()?.viewPath?.pathPart ?: return ResolveResult.UNRESOLVED
+
+        // NOTE(matklad): this is not necessary a module
```

Comment:
It is supposed to be general public. It may be surprising, but the following is valid:

```Rust
fn foo() {}

mod inner { use foo::{self}; }
```

That is, you can import something from any item, event from a function.